### PR TITLE
PortMidi Conan-ized

### LINKED ISSUE #1447: Extract and update PortMidi

### PR DESCRIPTION
**Pull request by crsib**
Resolves #1447 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: src/AudioIOBase.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ Paul Licameli split from AudioIO.cpp
 #endif
 
 #ifdef EXPERIMENTAL_MIDI_OUT
-#include "../lib-src/portmidi/pm_common/portmidi.h"
```
Comment (Paul-Licameli) [inline src/AudioIOBase.cpp:29]: Wherever there is `#include "../lib-src/..."` there is something wrong.

I count 14 such places in the source tree now (+1 that is commented out).  This removes four of those.

I'm glad to see that.  Will we see removal of the rest of them?

Comment (crsib) [reply src/AudioIOBase.cpp:29]: Will surely do :)

Comment (Paul-Licameli) [reply src/AudioIOBase.cpp:29]: Wait now, there is one inclusion of portmidi.h from NoteTrack.cpp that was not removed in this commit.  Did the next commit do it?

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]
Did you omit to change NoteTrack.cpp?

**Comment by Paul-Licameli** [comment]
I see NoteTrack.cpp also has
`#ifdef SONIFY`
and it's not defined in the default build.  I never did learn what this SONIFY thing was.  Other old team might confirm that it is no longer relevant.

So, that old #include wasn't breaking the build.  Still it could be changed consistently with the rest, or just deleted.  Then, less `#include "../lib/src..."` will remain to look at.

**Comment by crsib** [comment]
I fixed the include in `NoteTrack.cpp`

**Review by shoogle — APPROVED** [review]

**Pull request merged**